lib/ui/platform_dispatcher.dart

```diff
@@ -69,6 +69,19 @@ const double _kUnsetGestureSetting = -1.0;
 // See embedder.cc::kFlutterKeyDataChannel for more information.
 const String _kFlutterKeyDataChannel = 'flutter/keydata';
 
+class RootIsolateToken {
+  final int _rootIsolateId;
+  const RootIsolateToken._(this._rootIsolateId);
+
+  static RootIsolateToken? get instance {
+    final int rootIsolateId = __getRootIsolateId();
+    return rootIsolateId == 0 ? null : RootIsolateToken._(rootIsolateId);
```

Comment:
this will return a different instance each time (`RootIsolateToken.instance == RootIsolateToken.instance` is false) which I think would be surprising.

Reply:
done, switched to `late final`